### DISCUSSION TITLE
Inference time of YOLOv10 and YOLOv8

### DISCUSSION THREAD

**Discussion by kimhy365** [THU-MIG/yolov10#441] [General]
I normally use YOLOv5, v8, and I've been researching the recently released YOLOv10.
Specifically, I'm trying to learn the Nano model and apply it to the Jetson Nano.

1. I have FLOPs for 640 in YOLOv5, 8, and 10, but I don't have FLOPs for 416 and 224, is there a way to calculate them?
If it is a general classification model, I can calculate it with pytorch's profile, but if it includes post-processing.
I wonder if I can get it myself. (How to calculate)

2. I developed a v10 model with YOLOv8 and actually tried to infer it on Jetson Nano,
contrary to what the paper says, it was actually more time consuming.
Can you explain the results that contradict the paper?
In particular, the FLOPs of YOLOv8n and YOLOv10n were 8.1G and 8.2G, respectively, during the training process, and v10 was actually higher.

Thank you

**Comment by nicholasguimaraes** [comment]
I wonder the same. 

The GPU I'm testing is a Geforce RTX 2080.

When comparing yolov8n against yolov10n, version 8 actually take less time per iteration during inference (with a batch size of 1). 

yolov8n averaged 6.0ms per image. (not including preprocessing/postprocessing)     | allocated GPU memory 286mb 
yolov10n averaged 7.0ms per image. (not including preprocessing/postprocessing)   | allocated GPU memory 286mb 


Now when comparing yolov8L against yolov10L (both optimized with Tensorrt FP16),

TRT yolov8L averaged 20ms per image. (including preprocessing/postprocessing)     | allocated GPU memory 422mb
TRT yolov10L averaged 18ms per image. (including preprocessing/postprocessing)   | allocated GPU memory 386mb


So it seems that we might see the actual improvements by optimizing the .pt models on Tensorrt.

**Reply by kimhy365** [reply]
Thank you for the response.

My thoughts are as follows:

YOLOv10 seems to have additional elements in the model to eliminate NMS process in the post-processing.
In other words, it is designed to simplify the post-processing to reduce inference time.

However, since my dataset has only one object per image, the post-processing time for NMS etc. was relatively short. 
Therefore, I think YOLOv10's inference time is actually longer.

Thank you for your good comments.

**Reply by bshakya77** [reply]
Hello @nicholasguimaraes,
I am trying to benchmark various YOLO models on VisDrone dataset for inference time. Could you please provide the cmd or code that you used for calculating inference time (for single image or batch of images)?
Did you used the `predict `cmd provided by ultralytics?

`yolo predict model=runs/detect/train3/weights/last.pt source='../test2017/9999952_00000_d_0000001.jpg'`

I appreciate your feedback.

Regards,
Bijay

**Comment by kimhy365** [comment]
Thank you for the response.

My thoughts are as follows:

YOLOv10 seems to have additional elements in the model to eliminate NMS
process in the post-processing.
In other words, it is designed to simplify the post-processing to reduce
inference time.

However, since my dataset has only one object per image, the
post-processing time for NMS etc. was relatively short.
Therefore, I think YOLOv10's inference time is actually longer.

Thank you for your good comments.

2024년 10월 26일 (토) 오전 3:55, nicholasguimaraes ***@***.***>님이
작성:

> I wonder the same.
>
> The GPU I'm testing is a Geforce RTX 2080.
>
> When comparing yolov8n against yolov10n, version 8 actually take less time
> per iteration during inference (with a batch size of 1).
>
> yolov8n averaged 6.0ms per image. (not including
> preprocessing/postprocessing) | allocated GPU memory 286mb
> yolov10n averaged 7.0ms per image. (not including
> preprocessing/postprocessing) | allocated GPU memory 286mb
>
> Now when comparing yolov8L against yolov10L (both optimized with Tensorrt
> FP16),
>
> TRT yolov8L averaged 20ms per image. (including
> preprocessing/postprocessing) | allocated GPU memory 422mb
> TRT yolov10L averaged 18ms per image. (including
> preprocessing/postprocessing) | allocated GPU memory 386mb
>
> So it seems that we might see the actual improvements by optimizing the
> .pt models on Tensorrt.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/THU-MIG/yolov10/discussions/441#discussioncomment-11056052>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS7EEF2ZDJ2IUN4476HNS23Z5KHY5AVCNFSM6AAAAABONNPNASVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBVGYYDKMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

======================================================

충북대학교 산업인공지능연구센터 초빙교수 김현용


연락처 : 043) 249-1462 / 010-3023-3601

이메일 : ***@***.***

주   소 : 충북 청주시 청원구 오창읍 양청4길 45,

          충북대학교 오창캠퍼스 융합기술원 C655호

홈페이지 : https://indai.cbnu.ac.kr/

======================================================

**Comment by kimhy365** [comment]
I coded it as follows because I acquired a series of frames with opencv and
inferred them with the YOLO model.
The following code is not actual code, but it shows the procedure.
For more information, please visit the following website.
https://docs.ultralytics.com/modes/predict/

% Load yolo model
model = YOLO(model_path)

% Load 105 frame images: For-loop is required
img = cv2.imread(imfiles_path[i])

    % Calculate inference time:  Use the 100-time average after 5 warm-up
times
    t0 = time.time()
    results = model.predict(img, verbose=False)
    infer_time = (time.time() - t0) * 1000     % ms

I hope this will be helpful for your research.


2025년 1월 26일 (일) 오전 7:49, bshakya77 ***@***.***>님이 작성:

> Hello @nicholasguimaraes <https://github.com/nicholasguimaraes>,
> I am trying to benchmark various YOLO models on VisDrone dataset for
> inference time. Could you please provide the cmd or code that you used for
> calculating inference time (for single image or batch of images)?
> Did you used the predict cmd provided by ultralytics?
>
> yolo predict model=runs/detect/train3/weights/last.pt
> source='../test2017/9999952_00000_d_0000001.jpg'
>
> I appreciate your feedback.
>
> Regards,
> Bijay
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/THU-MIG/yolov10/discussions/441#discussioncomment-11954798>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS7EEF5FSPGQQIWG5VHZKOT2MQIGBAVCNFSM6AAAAABONNPNASVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOJVGQ3TSOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

======================================================

충북대학교 산업인공지능연구센터 초빙교수 김현용


연락처 : 043) 249-1462 / 010-3023-3601

이메일 : ***@***.***

주   소 : 충북 청주시 청원구 오창읍 양청4길 45,

          충북대학교 오창캠퍼스 융합기술원 C655호

홈페이지 : https://indai.cbnu.ac.kr/

======================================================
